### PR TITLE
Fix Paste will not replace selected content on Android #925

### PR DESCRIPTION
**Pull request by BryanValverdeU**
Seems like Android is not removing the selected range after pasting.
This PR fixes the issue

### CODE REVIEW

File: packages/roosterjs-editor-dom/lib/clipboard/extractClipboardEvent.ts [hunk 1]
```diff
@@ -1,39 +1,53 @@
...
+}
+
+function removeContents(rangeToRemove?: Range) {
+    return new Promise<void>(resolve => {
+        if (Browser.isAndroid && rangeToRemove) {
+            rangeToRemove.deleteContents();
+        }
+        resolve();
+    });
+}
```
Comment (BryanValverdeU) [inline packages/roosterjs-editor-dom/lib/clipboard/extractClipboardEvent.ts:53]: Not sure why all the file is getting modified.
Added this to remove the temp Range when is android

File: packages/roosterjs-editor-dom/lib/clipboard/extractClipboardEvent.ts [hunk 2]
```diff
@@ -1,39 +1,53 @@
...
+) {
+    const dataTransfer =
+        event.clipboardData ||
+        (<WindowForIE>(<unknown>(<Node>event.target).ownerDocument?.defaultView)).clipboardData;
+
+    if (dataTransfer.items) {
+        event.preventDefault();
+        extractClipboardItems(toArray(dataTransfer.items), options)
+            .then(callback)
+            .then(() => removeContents(tempRange));
```
Comment (BryanValverdeU) [inline packages/roosterjs-editor-dom/lib/clipboard/extractClipboardEvent.ts:40]: Also added this line

File: packages/roosterjs-editor-dom/lib/clipboard/extractClipboardEvent.ts [hunk 3]
```diff
@@ -1,39 +1,53 @@
...
+ *  image: Image file from the clipboard event
+ *  html: Html string from the clipboard event. When set to null, it means there's no HTML found from the event.
+ *   When set to undefined, it means can't retrieve HTML string, there may be HTML string but direct retrieving is
+ *   not supported by browser.
+ */
+export default function extractClipboardEvent(
+    event: ClipboardEvent,
+    callback: (clipboardData: ClipboardData) => void,
+    options?: ExtractClipboardEventOption,
+    tempRange?: Range
```
Comment (BryanValverdeU) [inline packages/roosterjs-editor-dom/lib/clipboard/extractClipboardEvent.ts:30]: This param was added too

File: packages/roosterjs-editor-dom/lib/clipboard/extractClipboardEvent.ts [hunk 4]
```diff
@@ -1,39 +1,53 @@
...
-        (<WindowForIE>(<unknown>(<Node>event.target).ownerDocument?.defaultView)).clipboardData;
-
-    if (dataTransfer.items) {
-        event.preventDefault();
-        extractClipboardItems(toArray(dataTransfer.items), options).then(callback);
-    } else {
-        extractClipboardItemsForIE(dataTransfer, callback, options);
-    }
-}
+import extractClipboardItems from './extractClipboardItems';
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-dom/lib/clipboard/extractClipboardEvent.ts:1]: can you check why the whole file is listed as changed? This makes we hard to find the changed place.

File: packages/roosterjs-editor-core/lib/corePlugins/CopyPastePlugin.ts [hunk 5]
```diff
@@ -130,7 +130,8 @@ export default class CopyPastePlugin implements PluginWithState<CopyPastePluginS
                 removeTempDiv: div => {
                     this.cleanUpAndRestoreSelection(div, range, false /* isCopy */);
                 },
-            }
+            },
+            this.editor?.getSelectionRange()?.cloneRange()
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-core/lib/corePlugins/CopyPastePlugin.ts:134]: why cloneRange() can fix this issue?

File: packages/roosterjs-editor-dom/lib/clipboard/extractClipboardEvent.ts (hunk 4 above)
Comment (BryanValverdeU) [reply packages/roosterjs-editor-dom/lib/clipboard/extractClipboardEvent.ts:1]: Opened another PR since I was not able to fix this.

**Comment by BryanValverdeU** [comment]
Opened another PR since I was not able to fix the whole file getting modified

**Pull request closed without merging**